### ISSUE TITLE
Have a button in the simple file picker to show/hide hidden files

### ISSUE THREAD

**Issue by rongxin-liu** [microsoft/vscode#136307]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.61.2 (6cba118ac49a1b88332f312a8f67186f7f3c1643)
- OS Version: Chrome 95.0.4638.54 (Official Build) (x86_64)

Steps to Reproduce:

1. Add glob patterns for excluding files and folders (e.g.,`.file_to_exclude` as shown in the attached screenshot)
2. Open Save As dialog from a file editor
3. Excluded files still listed in the save as dropdown menu

![Screen Shot 2021-11-02 at 14 30 07](https://user-images.githubusercontent.com/10591665/139924054-31102e5d-e7e3-4c33-abba-57401e346479.png)

cc @dmalan

**Comment by rongxin-liu** [comment]
I also tried adding `**/.file_to_exclude` to Search: Exclude, but am still seeing `.file_to_exclude` in the Save As dialog. Is it possible to hide files in the Save As dialog based on matching glob patterns specified in the settings somewhere?

**Comment by alexr00** [comment]
The simple file dialog, which is what you're seeing when you save as in Codesapces, functions as a replacement for the OS save dialog since we can't show the OS save dialog in this case. The OS save dialog doesn't respect the Files: Exclude setting, therefore the simple file dialog also doesn't respect that setting. 

I'm happy to hear other arguments for this though!

**Comment by dmalan** [comment]
@alexr00 just to argue in support of this one, in educational contexts using Codespaces (e.g., @cs50), we're finding that the simple file dialog's display of dotfiles is a bit confusing for first-time programmers, since they don't see those same dotfiles  in the file explorer or terminal window (based on our `.devcontainer.json` and shell settings). As a result, in the worst case, if students accidentally save some source file in a dotfolder, they can't easily find it again thereafter, effectively losing work. (We can help them troubleshoot, of course, but it does increase FAQs.)

Given that the simple file dialog is tied to Codespaces anyway, might it be compelling for it to respect that same **Files: Exclude** setting (or an equivalent)?

Thank you!

CC @lostintangent

**Comment by alexr00** [comment]
@dmalan the simple file dialog can also be turned on as the default file dialog with a setting too, and it is the file picker for the Remote WSL, Remote SSH, and Remote Containers extensions. Hiding files in all those setups means there's no way to open those files.

That said, OS file pickers typically have "show hidden files" button/checkbox. I'd be happy to have a "show hidden files" button that functions similarly, with the added benefit of showing/hiding files from `Files: Exclude` instead of just dotfiles.

**Comment by dmalan** [comment]
Hi @alexr00, oh sure, that would work great, if the button/checkbox could default to hiding dotfiles and dotfolders in that case? Many thanks!

**Comment by dmalan** [comment]
Hi @alexr00, just wanted to surface this one again, as it's tended to come up with students. Any chance this is still on the road map? Thank you!

**Comment by jrussell9000** [comment]
Also bumping this......it's a bit obnoxious when using WSL to have to scroll past so many dot-lead directory names.

**Comment by amlamarra** [comment]
Same! I use VScode to SSH to my VM and would love to be able to hide the dotfiles.

**Comment by alexr00** [comment]
Adding:

![Image](https://github.com/user-attachments/assets/af644ab3-927b-4b19-9ec7-fd6d9243d14b)

**Comment by amlamarra** [comment]
Much appreciated!

Will we be able to set a default behavior in the settings?

**Comment by alexr00** [comment]
No, no setting at the moment. Let's try it out for a couple months and see if we actually need a setting.

**Comment by garysassano** [comment]
So how do you hide the dot folders from file picker if you work with WSL2?

**Comment by alexr00** [comment]
![Image](https://github.com/user-attachments/assets/7cc09111-3a6a-4cf7-b1dc-08d582503545)

It's remembered per workspace.

**Comment by garysassano** [comment]
When is it coming to the stable release?

**Comment by garysassano** [comment]
When is this feature coming to the stable release?
